### PR TITLE
fix(pie-monorepo): DSW-1434 add missing component imports

### PR DESCRIPTION
**Pull request by xander-marjoram**
## Describe your changes (can list changeset entries if preferable)
- Add missing component imports
- Update READMEs with new instructions
- Add/update component imports in stories
- Update generator to include README and story changes
- Move pie component dependencies from `devDependencies` to `dependencies` for modal and cookie-banner
- Set `@justeattakeaway` packages as externals (exclude from bundle) for modal and cookie banner

## Author Checklist (complete before requesting a review)
- [x] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests
- [ ] If it is a `PIE Docs` change, I have reviewed the Docs site preview
- [ ] If it is a component change, I have reviewed the Storybook preview
- [ ] If there are visual test updates, I have reviewed them properly before approving

### Reviewer 1
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### Reviewer 2
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### CODE REVIEW

**Comment by xander-marjoram** [comment]
I'm not the biggest fan of having to disable eslint in lots of the files, but I imagine that most consumers of the components will only have to import the module to register the component.

**Comment by xander-marjoram** [comment]
Not yet ready for review, I just want to check the visual tests still work

**Review by jamieomaguire — APPROVED** [review]

File: packages/components/pie-cookie-banner/src/index.ts [hunk 1]
```diff
@@ -1,10 +1,20 @@
 import {
     LitElement, html, unsafeCSS, TemplateResult, nothing,
 } from 'lit';
-import { defineCustomElement } from '@justeattakeaway/pie-webc-core';
 import { property, state, queryAll } from 'lit/decorators.js';
 import { repeat } from 'lit/directives/repeat.js';
+
+import '@justeattakeaway/pie-button';
```
Comment (ashleynolan) [inline packages/components/pie-cookie-banner/src/index.ts:7]: Out of interest – what's the bundle size difference between the before/after of this change? (would be useful to know for when we get back onto the bundling in the next couple of weeks)

**Review by ashleynolan — APPROVED** [review]
🎉

**Review by jamieomaguire — APPROVED** [review]

File: packages/components/pie-modal/vite.config.js [hunk 2]
```diff
@@ -1,3 +1,20 @@
...
+                index: 'src/index.ts',
+                react: 'src/react.ts',
+            },
+            formats: ['es'],
+        },
+        rollupOptions: {
+            external: [
+                'react',
+                /^lit/,
+                /@justeattakeaway/
```
Comment (ashleynolan) [inline packages/components/pie-modal/vite.config.js:16]: I'm guessing it'd just make sense to add this into the shared vite config potentially as a follow up (as all components should exclude these?)

Comment (xander-marjoram) [reply packages/components/pie-modal/vite.config.js:16]: Yes, that is the logical progression but I didn't want to do too much at once in this PR

**Comment by fernandofranca** [comment]
With moving deps from devDependencies to dependencies aren't we risking to bundle code twice if a consumer imports both cookie banner and button in an app?

**Comment by xander-marjoram** [comment]
@fernandofranca I tried it in the vanilla Aperture app and it actually behaves extremely nicely. Adding `@justeattakeaway/` as `external` for modal and cookie banner means the button (and others) won't be bundled inside them. We also don't see a duplicate registration error, presumably because the browser is clever enough to only load that bundle once! Looking at the Network tab seems to confirm this, as the button bundle is only downloaded once even though I import both components.

```javascript
import '@justeattakeaway/pie-button';
import '@justeattakeaway/pie-modal';
```

**Review by ashleynolan — APPROVED** [review]

File: packages/components/pie-cookie-banner/src/index.ts (hunk 1 above)
Comment (xander-marjoram) [reply packages/components/pie-cookie-banner/src/index.ts:7]: | Component | Before | After |
| --- | --- | --- |
| `pie-modal` | 30.9kB | 22.7kB |
| `pie-cookie-banner` | 64.7kB |15.7kB |

It's also worth noting that before this PR, the cookie banner bundle also included the modal's dialog polyfill. Now it is only bundled as part of the modal.

**Review by fernandofranca — APPROVED** [review]

**Pull request merged**